### ISSUE TITLE
pyinfra doesn't ask for sudo password when config.SUDO or sudo = True and get_pty = True

### ISSUE THREAD

**Issue by jmpolom** [Fizzadar/pyinfra#766]
## Describe the bug
```
from pyinfra import config
from pyinfra.operations import server

config.SUDO = True

server.shell(
    name = 'Try it with shell',
    commands = 'systemctl --user --machine=dwspectrum@.host daemon-reload',
    get_pty = True
)
```

or 

```
from pyinfra import config
from pyinfra.operations import server

server.shell(
    name = 'Try it with shell',
    commands = 'systemctl --user --machine=dwspectrum@.host daemon-reload',
    get_pty = True,
    sudo = True
)
```

Either of the above will result in pyinfra not asking for a sudo password, which is different from the behavior when `get_pty` is not specified and pyinfra asks for a sudo password if one is not supplied via a kwarg or configuration parameter. This seems like buggy inconsistent behavior.

Example output when it doesn't prompt for password:
```
~/homenet/router/pyinfra master ?3 ❯ pyinfra -vvv hosts.py test_systemd.py
--> Loading config...
--> Loading inventory...

--> Connecting to hosts...
    [archon] Connected

--> Preparing operations...
    Loading: test_systemd.py
    [archon] Ready: test_systemd.py

--> Proposed changes:
    Groups: routers / hosts
    [archon]   Operations: 1   Commands: 1

--> Beginning operation run...
--> Starting operation: Try it with shell
[archon] >>> sudo -H -n sh -c 'systemctl --user --machine=dwspectrum@.host daemon-reload'
[archon] sudo: a password is required
    [archon] Error
--> pyinfra error: No hosts remaining!
```

To get around this issue, either `config.USE_SUDO_PASSWORD` needs to be set to `True`/the password value, or the equivalent kwarg can be passed to the operation with the same values.

## To Reproduce
Add `get_pty=True` to an operation.

## Expected behavior
pyinfra prompts for sudo password as it does when `get_pty` is default/False.

## Meta
+ Include output of `pyinfra --support`.
```
--> Support information:
    System: Darwin
      Platform: macOS-12.2.1-arm64-arm-64bit
      Release: 21.3.0
      Machine: arm64
    pyinfra: v1.6.3
    Executable: /Users/jon/homenet/router/pyinfra/venv/bin/pyinfra
    Python: 3.9.10 (CPython, Clang 13.0.0 (clang-1300.0.29.3))
```
+ How was pyinfra installed (source/pip)? git, master branch
+ Include pyinfra-debug.log (if one was created)
+ Consider including output with `-vv` and `--debug`.

**Comment by Fizzadar** [comment]
Replicated this and `get_pty` means output lines have `\r` at the end - I've added a fix (+test) for this in https://github.com/Fizzadar/pyinfra/commit/f5cb24e5ae99f457fc7d725d9d087c2fdc5b7ed7.

**Comment by Fizzadar** [comment]
Now released in `v1.7`!
